### ISSUE TITLE
Likelyhood fitting

### ISSUE THREAD

**Issue by lrnv** [ChaoranHu/coga#3]
Hey, 

Do you now if there exist somewhere an algorithm to fit gamma convolutions to data ? For exemple with a fixed number of gamma random variable.

**Comment by ChaoranHu** [comment]
Sorry for the very late reply, @lrnv!
Just in case you still need this information, we do not have an analytical formula for this question. So, the optimization can be done by common numerical way.

--C.Hu

**Comment by lrnv** [comment]
Hey ! Thanks for reaching out. Funny enough, I forgot about the very existence of this issue... but not about the problem itself: I ended up extensively answering the question myself and this made the core of my PhD.

To put it in a nutshell, common maximization of the coga likelihood will not work due to instabilities. I developed a few better estimation methods for these models, together with a lot of theoretical analysis about the class of (multivariate) generalized gamma convolutions, with a few outputs that might be of interest to you if you still study the Thorin classes: 

- Paper 1: [Estimation of multivariate generalized gamma convolutions through Laguerre expansions.](https://projecteuclid.org/journals/electronic-journal-of-statistics/volume-15/issue-2/Estimation-of-multivariate-generalized-gamma-convolutions-through-Laguerre-expansions/10.1214/21-EJS1918.short)
- Paper 2: [Estimation of high dimensional Gamma convolutions through random projections](https://arxiv.org/abs/2203.13741)
- Julia code: [lrnv/ThorinDistributions.jl](https://github.com/lrnv/ThorinDistributions.jl)
- Vulgarization article: [Parametric divisibility of stochastic losses](https://arxiv.org/abs/2210.06034)

Sadly, I was not able to leverage `R::coga` for precision issues (details in paper 1), and I had to reimplement it (along a lot of other things) in Julia to get acess to higher precision computations. But nevertheless, I thought this might be interesting for you.
